### PR TITLE
Product page

### PR DESCRIPTION
**Pull request by nagwan**

### CODE REVIEW

File: frontend/src/components/partials/Footer/Footer.js [hunk 1]
```diff
@@ -2,8 +2,6 @@ import React from "react"
 import "./footer.sass"
 export default function Footer(){
     return(
-        <footer className="w-100 bg-main-800">
-
-        </footer>
+        <footer className="w-100 bg-main-800"></footer>
```
Comment (MoatasemEbrahim) [inline frontend/src/components/partials/Footer/Footer.js:5]: I think empty elements should be self closed
`<footer className="w-100 bg-main-800" />`

File: frontend/src/pages/Home.js [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
-                        <ProductCard product={product}
-                        />
-                    </Link>
-                </div>
-            ))
-        } </div>
+        <div className="w-100">
+            {isLoading ? <Loader /> :
+
+                (error ? <h3 className="txt-color-danger-500 fs-16">{error}</h3> : <div className="container d-flex flex-wrap" > {
```
Comment (MoatasemEbrahim) [inline frontend/src/pages/Home.js:20]: avoid using nested ternary operator (use if statements to assign the element should be rendered to a variable before the return() and use the variable in the JSX)

File: frontend/src/pages/Product.js [hunk 3]
```diff
@@ -1,69 +1,78 @@
...
-                </div>
-                <div className="col-12 col-md-6 col-lg-3">
-                    <p className="fs-20 font-primary-bold txt-color-main-900">${product.price}</p>
+                    <>
+                        <div className="d-flex m-y-15">
+                            <div className="col">
+                                <Link className="btn btn-link no-decoration txt-color-main-500" to="/">GO Back</Link>
+                            </div>
+                        </div>
+                        {error ? <h3 className="txt-color-danger-500 fs-16">{error}</h3> :
```
Comment (MoatasemEbrahim) [inline frontend/src/pages/Product.js:28]: what does fs-16 className represent ??

File: frontend/src/store/products/actions.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import { api } from "../../utils/Api"
+
+/**
+ * products list actions types
+ */
+export const PRODUCT_LIST_REQUEST = "PRODUCT_LIST_REQUEST"
```
Comment (MoatasemEbrahim) [inline frontend/src/store/products/actions.js:6]: Why you don't define action types as constants in separate files?

**Review by MoatasemEbrahim — APPROVED** [review]
General tip:
Leave an empty line at the end of every file as git recommends, to be able to merge new code easily in old files and avoid fake conflicts

File: frontend/src/store/products/actions.js (hunk 4 above)
Comment (nagwan) [reply frontend/src/store/products/actions.js:6]: i thought there is no need for

**Pull request merged**